pages/games/[hash].js

```diff
@@ -0,0 +1,52 @@
+import styles from "../../styles/Game.module.css";
+
+function Game({ data }) {
+  return (
+    <div>
+      Game page {data.game_hash}
+      <div className={styles.board}>
+        <div className={styles.fleet}>
+          {data.ships.map((ship, i) => (
+            <div
+              key={i}
+              className={styles.square}
+              style={formatGridAttributes(ship.grid_attributes)}
+            >
+              <div className={styles.ship}>
+              </div>
+            </div>
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export async function getServerSideProps(context) {
+  const res = await fetch(`http://localhost/api/games/${context.params.hash}`);
+  const data = await res.json();
+
+  return { props: { data } };
+}
+
+const formatGridAttributes = (str) => {
```

Comment:
You can extract these functions and write unit tests. We generally use [jest](https://jestjs.io/).

Reply:
It could be done in another small PR ;)

Reply:
Ok. It will be done in future PR